solr/core/src/java/org/apache/solr/cloud/overseer/CollectionMutator.java

```diff
@@ -160,6 +164,38 @@ public ZkWriteCommand modifyCollection(final ClusterState clusterState, ZkNodePr
     }
   }
 
+  /**
+   *  Update the state/leader of replicas in state.json from PRS data
+   */
+  public static DocCollection updateReplicasFromPrs(DocCollection coll, PerReplicaStates prs) {
+    //we are disabling PRS. Update the replica states
+    Map<String, Slice> modifiedSlices = new LinkedHashMap<>();
+    coll.forEachReplica((s, replica) -> {
+      PerReplicaStates.State prsState = prs.states.get(replica.getName());
+      if (prsState != null) {
+        if (prsState.state != replica.getState()) {
```

Comment:
This if statement and the next one look like they are almost doing the same thing, is there a way to simplify this logic so that we are only getting/setting from modifiedSlices once?

Reply:
I thought about it. I thought this would make it simpler. suggestions are welcome

Reply:
Yea, I don't think I have any better ways.